frontstage/views/passwords.py

```diff
@@ -24,6 +25,46 @@ def forgot_password():
         email_address = request.form.get('email_address')
         post_data = {"email_address": email_address}
 
+        try:
+            token_url = app.config['ONS_TOKEN']
+
+            data = {
+                'grant_type': 'reset_password',
+                'client_id': app.config['RAS_FRONTSTAGE_CLIENT_ID'],
+                'client_secret': app.config['RAS_FRONTSTAGE_CLIENT_SECRET'],
+                'username': email_address,
+            }
+            headers = {
+                'Accept': 'application/json',
+                'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
+            }
+
+            oauth2_response = requests.post(url=token_url, data=data,
+                                            headers=headers, auth=(app.config['RAS_FRONTSTAGE_CLIENT_ID'],
+                                                                   app.config['RAS_FRONTSTAGE_CLIENT_SECRET']))
+            # Check to see that this user account is not locked
+            if oauth2_response.status_code == 401:
+                oauth2Error = json.loads(oauth2_response.text)
+                if 'Unauthorized user credentials' in oauth2Error['detail']:
+                    logger.warning('Email address is not registered')
+                    template_data = {"error": {"type": {"Email address is not registered"}}}
```

Comment:
This template data isn't used anywhere currently, this is an existing problem with the original form however rather than something that's been added here I think.